### DISCUSSION TITLE
Apply different weights in loss

### DISCUSSION THREAD

**Discussion by DevergneTimothee** [deepmodeling/deepmd-kit#358] [Q&A] [answered]
Hello, I am using deepmd, on a system with many different atoms. I would like to know if it is possible to apply a weight in the loss of the atomic forces : imagine I have three atoms (A1,A2,A3) of type 1 and 2 such that A1 is of type 1 and A2, A3 of type 3. What i would like to do, is that in the force loss, the force acting on A1 has twice the importance of the force acting on A2 or A3, so that the error on the force for type 2 doesn't dominate the error on the force for type 1.
I saw in the code that there is a feature called "atom_pref" that might do what i want to do, but there is no documentation about it... Is it possible to use it ?  Or is there a way to modify the code so that it does what i want ?

Best

**Comment by njzjz** [comment, marked as answer]
Yes, you can add `atom_pref.npy` under the directory of data set (where `energy.npy` is). The shape of this numpy file is (Nframes, Natoms).

**Reply by njzjz** [reply]
@DevergneTimothee And you need to use `start_pref_pf` and `limit_pref_pf` instead of `start_pref_f` and `limit_pref_f`.

**Reply by DevergneTimothee** [reply]
Thanks a lot, I'm going to try this out then.
